tests/pytorch/test_pytorch_model_export.py

```diff
@@ -440,6 +441,211 @@ def predict(self, context, model_input):
         decimal=4)
 
 
+def test_load_pyfunc_loads_torch_model_using_pickle_module_specified_at_save_time(
+        module_scoped_subclassed_model, model_path):
+    mlflow.pytorch.save_model(
+        path=model_path,
+        pytorch_model=module_scoped_subclassed_model,
+        conda_env=None)
+
+    custom_pickle_module_name = "test pickle module"
+
+    pyfunc_conf = _get_flavor_configuration(model_path=model_path, flavor_name=pyfunc.FLAVOR_NAME)
+    model_data_path = os.path.join(model_path, pyfunc_conf[pyfunc.DATA])
+    assert os.path.exists(model_data_path)
+    assert mlflow.pytorch._PICKLE_MODULE_INFO_FILE_NAME in os.listdir(model_data_path)
+    with open(
+            os.path.join(model_data_path, mlflow.pytorch._PICKLE_MODULE_INFO_FILE_NAME), "w") as f:
+        f.write(custom_pickle_module_name)
+
+    import_module = importlib.import_module
+    def validate_pickle_module_import(module_name):
+        if module_name == mlflow.pytorch.__name__:
+            return import_module(module_name)
+        else:
+            assert module_name == custom_pickle_module_name
+            return module_name
+
+    def validate_torch_load(*args, **kwargs):
+        # pylint: disable=unused-argument
+        assert kwargs["pickle_module"] == custom_pickle_module_name
+
+    with mock.patch("importlib.import_module") as import_mock,\
+            mock.patch("torch.load") as torch_load_mock:
+        import_mock.side_effect = validate_pickle_module_import
+        torch_load_mock.side_effect = validate_torch_load
+        pyfunc.load_pyfunc(model_path)
+
+
+def test_load_model_loads_torch_model_using_pickle_module_specified_at_save_time(
+        module_scoped_subclassed_model):
+    artifact_path = "pytorch_model"
+    with mlflow.start_run():
+        mlflow.pytorch.log_model(
+            artifact_path=artifact_path,
+            pytorch_model=module_scoped_subclassed_model,
+            conda_env=None)
+        run_id = mlflow.active_run().info.run_uuid
+    model_path = tracking.utils._get_model_log_dir(artifact_path, run_id)
+
+    custom_pickle_module_name = "test pickle module"
+
+    pyfunc_conf = _get_flavor_configuration(model_path=model_path, flavor_name=pyfunc.FLAVOR_NAME)
+    model_data_path = os.path.join(model_path, pyfunc_conf[pyfunc.DATA])
+    assert os.path.exists(model_data_path)
+    assert mlflow.pytorch._PICKLE_MODULE_INFO_FILE_NAME in os.listdir(model_data_path)
+    with open(
```

Comment:
the same as above. Also, could we merge the two tests into one? Seems like there is a lot of repetition.

Reply:
Makes sense, thanks!